test/topology_custom/test_mv_topology_change.py

```diff
@@ -193,3 +194,46 @@ async def test_mv_write_to_dead_node(manager: ManagerClient):
     # will be held for long time until the write timeouts.
     # Otherwise, it is expected to complete in short time.
     await manager.remove_node(servers[0].server_id, servers[-1].server_id, timeout=30)
+
+# Reproduces https://github.com/scylladb/scylladb/issues/21492
+# In this test we perform a write with a view update while the replication factor
+# of the keyspace is being changed and the corresponding base tablet finishes
+# the migration at a different time the view tablet does.
+@pytest.mark.asyncio
+@pytest.mark.parametrize("delayed_replica", ["base", "view"])
+@skip_mode('release', 'error injections are not supported in release mode')
+async def test_mv_rf_change(manager: ManagerClient, delayed_replica: str):
+    servers = await manager.servers_add(2)
```

Comment:
I think the testcase needs to be extended with multiple DCs. See `test_multidc_alter_tablets_rf`

Reply:
While I don't see how different DCs could currently interact in `get_view_natural_endpoint`, this could help with noticing if some future patch changed that, so I'll add it in the next version